x-pack/legacy/plugins/apm/public/components/app/ServiceNodeOverview/index.tsx

```diff
@@ -46,24 +45,27 @@ const ServiceNodeOverview = () => {
     [serviceName]
   );
 
-  const { data: items } = useFetcher(() => {
-    if (!serviceName || !start || !end) {
-      return;
-    }
-    return callApmApi({
-      pathname: '/api/apm/services/{serviceName}/serviceNodes',
-      params: {
-        path: {
-          serviceName
-        },
-        query: {
-          start,
-          end,
-          uiFilters: JSON.stringify(uiFilters)
-        }
+  const { data: items = [] } = useFetcher(
+    callApmApi => {
+      if (!serviceName || !start || !end) {
+        return undefined;
```

Comment:
I think you don't have to manually return undefined here, just check if all variables are filled and return the `callApmApi`, otherwise it returns `undefined`.

Reply:
Aye, trying to be explicit here. To me `return;` just looks odd if you expect the function to return _something_ (including `undefined`).